Storage/SQL/SQLiteBookmarksBase.swift

```diff
@@ -35,4 +45,21 @@ public class SQLiteBookmarks: BookmarksModelFactorySource {
         
         return self.db.queryReturnsResults(sql, args: args)
     }
+
+    public func getURLForKeywordSearch(keyword: String) -> Deferred<Maybe<String>> {
+        let sql = "SELECT bmkUri FROM \(TableBookmarksBuffer) WHERE " +
```

Comment:
Sure thing. Since you're going to be modifying a SQL VIEW instead of a SQL TABLE you won't need to do an ALTER TABLE.

As Richard mentioned, we already have the `keyword` field in the bookmark tables [1] but we don't have them part of the `ViewBoomarksBufferOnMirror` SQL VIEW [2]. Since VIEWs are read-only and are queries on top of existing tables, instead of altering it, you'll need to just `DROP` the VIEW and recreate it. We've done this in a few places already that you can use as reference [3]. As far as where the migration code should live, you'll want to add another `if` case to the `updateTable` in `BrowserTable` and update the `DefaultVersion` to DefaultVersion+1 along with an associated bug number [4]

[1] https://github.com/mozilla-mobile/firefox-ios/blob/master/Storage/SQL/BrowserTable.swift#L299
[2] https://github.com/mozilla-mobile/firefox-ios/blob/master/Storage/SQL/BrowserTable.swift#L380
[3] https://github.com/mozilla-mobile/firefox-ios/blob/master/Storage/SQL/BrowserTable.swift#L778
[4] https://github.com/mozilla-mobile/firefox-ios/blob/master/Storage/SQL/BrowserTable.swift#L110